.github/workflows/tests.yml

```diff
@@ -0,0 +1,50 @@
+
+name: CI-Tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  BuildTestContainer:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build Test Image
+        uses: docker/build-push-action@v1.1.0
+        with:
+          username: xinsnake
```

Comment:
I think we need to think about what to do here.

Reply:
Reason to build and push containers every time:

By default GitHub action does not allow you to run a container in background with `docker run`, and you can only do it by using [Service Containers](https://help.github.com/en/actions/configuring-and-managing-workflows/about-service-containers), and the service container has to be a container hosted on DockerHub because you cannot pass in credentials to use a private repository.